### ISSUE TITLE
Provide a way for `dotnet sln add` to Pipe Input

### ISSUE THREAD

**Issue by aolszowka** [dotnet/sdk#11558]
`dotnet sln add` allows for you to specify multiple projects on the command line, but you are limited to 32,768 characters per command line.

It would be nice if `dotnet sln add` supported an input file of some sort to take multiple arguments.

Our use cases involves attempting to create solutions with several thousand project files via composition of existing `sln` files. The current implementation invokes `dotnet sln add` per project and is extremely slow.

It is believed that if we could specify an input file of some sort to this command that we could drastically speed up creation of these solution files.

Below is a Powershell snippet that shows the functions in question.

```powershell
function Compose-Solution([string]$SolutionPath, [string[]]$TargetSolutions)
{
    $SOLUTION_DIRECTORY = [System.IO.Path]::GetDirectoryName($SolutionPath)
    $SOLUTION_NAME = [System.IO.Path]::GetFileNameWithoutExtension($SolutionPath)

    if ([System.IO.File]::Exists($SolutionPath))
    {
        Remove-Item $SolutionPath
    }

    &dotnet.exe new sln -n $SOLUTION_NAME -o $SOLUTION_DIRECTORY

    $AllProjects = Get-AllProjects -TargetSolutions $TargetSolutions

    foreach ($project in $AllProjects)
    {
        &dotnet.exe sln $SolutionPath add $project
    }
}

function Get-AllProjects([string[]]$TargetSolutions)
{
    foreach ($solution in $TargetSolutions)
    {
        $AllProjects += Get-ProjectsInSolution($solution)
    }

    $AllProjects | Select-Object -Unique
}

function Get-ProjectsInSolution([string]$TargetSolution)
{
    if ([System.IO.File]::Exists($TargetSolution))
    {
        $solutionDirectory = [System.IO.Path]::GetDirectoryName($TargetSolution)
        $solutionListing = &dotnet.exe sln $TargetSolution list | Select-Object -Skip 2

        foreach ($project in $solutionListing )
        {
            $relativePath = [System.IO.Path]::Combine($solutionDirectory, $project)
            [System.IO.Path]::GetFullPath($relativePath)
        }
    }
}
```

**Comment by marcpopMSFT** [comment]
This has been fixed since 5.0.2xx with the introduction of system.commandline as we now support response files with the @ argument.

**Comment by marcpopMSFT** [comment]
https://learn.microsoft.com/en-us/dotnet/standard/commandline/syntax#response-files
